_rules/inline-link-in-paragraph-has-distinguishable-border-36f116.md

```diff
@@ -0,0 +1,159 @@
+---
+id: 36f116
+name: Inline link has distinguishable border
+rule_type: atomic
+description: |
+  This rule checks that inline links have a border that distinguishes them from the surrounding text
+accessibility_requirements:
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgments:
+  authors:
+    - Brian Bors
+    - Carlos Duarte
+  previous authors:
+    - Wilco Fiers
+---
+
+## Applicability
+
+This rule applies to any [visible][] HTML element that is a [semantic link][], where the closest element (including itself) that generates an [inline box][] is part of a [line box][] that has [inline boxes][inline box] generated from an element containing [visible][] [text nodes][text node] that do not have a [semantic link][] as a [descendant][] in the [flat tree][].
+
+**Note:** The test targets are visible link elements that are part of a block of content that is displayed inline.
+
+## Expectation
+
+Each target element has a [visible][] [border](https://drafts.csswg.org/css-backgrounds/#borders).
```

Comment:
```html
<p>Read about WAI on the <u><a style="text-decoration:none" href="">WAI webpage</a></u>.</p>
```

This rule would fail this because it isn't the link, but the parent of a link that has the style applied to it. Similarly a child of the link could have the border applied to it. This can't be limited to the link. The same is true for some of these other rules.

Reply:
This rule, in particular, would fail because there are no borders (`<u>` renders with a text-decoration-line equal to underline, not any kind of border). So it fails, and should fail.

It would pass "Inline link has distinguishable style not based on color alone". It would pass because the `<a>` element would have a distinguishing style from the `<u>` (based on the text-decoration-line property).